src/Model.js

```diff
@@ -0,0 +1,215 @@
+		this.namespace = '';
+		this.mainPage = false;
+	}
+
+	/**
+	 * Initialize the model with the given content and configuration
+	 * This is done after the class is instantiated, when we are certain
+	 * that the DOM loaded with all gadgets ready.
+	 *
+	 * @param {jQuery} $content jQuery object representing the content
+	 * @param {Object} config Configuration object
+	 * @param {string} [config.lang='en'] Interface language used
+	 * @param {string} [config.namespace] Namespace of the current article
+	 * @param {boolean} [config.mainPage] Whether the current page is the
+	 *  wiki's main page
+	 * @fires Model#initialized
+	 */
+	initialize( $content, config ) {
+		this.$contentWrapper = $content;
+		this.lang = config.lang || 'en';
+		this.namespace = config.namespace || '';
+		this.mainPage = !!config.mainPage;
+
+		this.initialized = true;
+
+		/**
+		 * Initialization event
+		 *
+		 * @event Model#initialized
+		 */
+		this.emit( 'initialized' );
+	}
+
+	/**
+	 * Cache the original content we are about to replace
+	 *
+	 * @param {jQuery} [$content=this.$contentWrapper] A jQuery
+	 *  object to cache. If not given the $contentWrapper will
+	 *  be cached. In most cases this should remain empty,
+	 *  as $contentWrapper contains the original content before
+	 *  it is replaced.
+	 */
+	cacheOriginal( $content = this.$contentWrapper ) {
+		$content = $content || this.$contentWrapper;
+		this.$originalContent = $content.clone( true, true );
+	}
+
+	/**
+	 * Get the content wrapper for the current page
+	 *
+	 * @return {jQuery} Content wrapper
+	 */
+	getContentWrapper() {
+		return this.$contentWrapper;
+	}
+
+	/**
+	 * Get the original content of the article
+	 *
+	 * @return {jQuery} Original content
+	 */
+	getOriginalContent() {
+		return this.$originalContent;
+	}
+
+	/**
+	 * Check if the current page is valid
+	 * for the system to display.
+	 *
+	 * @return {boolean} Page is valid
+	 */
+	isValidPage() {
+		return !!(
+			// Has the needed parser content
+			this.getContentWrapper().length &&
+			// Is in the main namespace
+			this.namespace === '' &&
+			// Is not main page
+			!this.mainPage
+		);
+	}
+
+	/**
+	 * Check whether the model was initialized
+	 *
+	 * @return {boolean} Model was initialized
+	 */
+	isInitialized() {
+		return this.initialized;
+	}
+
+	/**
+	 * Check whether the system is enabled on this page
+	 *
+	 * @return {boolean} System is enabled
+	 */
+	isEnabled() {
+		return this.enabled;
+	}
+
+	/**
+	 * Check whether the system is currently active
+	 *
+	 * @return {boolean} System is active
+	 */
+	isActive() {
+		return this.active;
+	}
+
+	/**
+	 * Get the current state of the system.
+	 * Possible values are 'pending', 'ready' or 'error'.
+	 *
+	 * @return {string} State of the system
+	 */
+	getState() {
+		return this.state;
+	}
+
+	/**
+	 * Toggle the active state of the system
+	 *
+	 * @fires Model#active
+	 * @param {boolean} [isActive] System is active
+	 */
+	toggleActive( isActive ) {
```

Comment:
Instead of `isActive` maybe `newState`, to make it clearer that it's the *target* state, not the current state?